client/src/Movies/Movie.js

```diff
@@ -1,57 +1,51 @@
-import React, { useState, useEffect } from 'react';
+import React from 'react';
 import axios from 'axios';
+import MovieCard from './MovieCard'
+export default class Movie extends React.Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      movie: null,
+    };
+  }
 
-const Movie = (props) => {
-  const [movie, setMovie] = useState({});
- 
-  useEffect(() => {
-    const id = 1;
-    // change ^^^ that line and grab the id from the URL
-    // You will NEED to add a dependency array to this effect hook
-
-       axios
-        .get(`http://localhost:5000/api/movies/${id}`)
-        .then(response => {
-          setMovie(response.data);
-        })
-        .catch(error => {
-          console.error(error);
-        });
-
-  },[]);
-  
-  // Uncomment this only when you have moved on to the stretch goals
-  // const saveMovie = () => {
-  //   const addToSavedList = props.addToSavedList;
-  //   addToSavedList(movie)
-  // }
+  componentDidMount() {
+    this.fetchMovie(this.props.match.params.id)
+  }
 
-  if (!movie) {
-    return <div>Loading movie information...</div>;
+  componentWillReceiveProps(newProps){
+    if(this.props.match.params.id !== newProps.match.params.id){
+      this.fetchMovie(newProps.match.params.id);
+    }
   }
 
-  const { title, director, metascore, stars } = movie;
-  return (
-    <div className="save-wrapper">
-      <div className="movie-card">
-        <h2>{title}</h2>
-        <div className="movie-director">
-          Director: <em>{director}</em>
-        </div>
-        <div className="movie-metascore">
-          Metascore: <strong>{metascore}</strong>
-        </div>
-        <h3>Actors</h3>
+  fetchMovie = (id) => {
+    // change this line to grab the id passed on the URL
+    axios
+      .get(`http://localhost:5000/api/movies/${id}`)
+      .then(response => {
+        this.setState(() => ({ movie: response.data }))
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
 
-        {stars.map(star => (
-          <div key={star} className="movie-star">
-            {star}
-          </div>
-        ))}
+  saveMovie = () => {
+    const addToSavedList = this.props.addToSavedList;
+    addToSavedList(this.state.movie)
+  }
+  
+  render() {
+    if(!this.state.movie) {
+      return <div>Grabbing Movies</div>
+    }
+    
```

Comment:
You don't need render in a functional component only class components. We haven't gone over those yet. But i like that you are trying new things!